libqtile/bar.py

```diff
@@ -175,34 +177,64 @@ def __init__(self, widgets, size, **config):
         self.size_calculated = 0
         self._configured = False
 
+        if isinstance(self.margin, int):
+            self.margin = [self.margin] * 4
+
+        if isinstance(self.border_width, int):
+            self.border_width = [self.border_width] * 4
+
+        self._initial_margin = self.margin[:]
+        self.struts = [0, 0, 0, 0]
+        self._add_strut = False
+
         self.queued_draws = 0
         self.future = None
 
     def _configure(self, qtile, screen):
-        Gap._configure(self, qtile, screen)
+        # We only want to adjust margin sizes once unless there's a new strut
+        if not self._configured or self._add_strut:
+            Gap._configure(self, qtile, screen)
+
+            if sum(self._initial_margin) or sum(self.border_width) or self._add_strut:
+
+                try:
+                    # Check if colours are valid but don't convert to rgba here
+                    if isinstance(self.border_color, list) and len(self.border_color) == 4:
+                        [rgb(col) for col in self.border_color]
+                    else:
+                        rgb(self.border_color)
+                        self.border_color = [self.border_color] * 4
+                except (ValueError, TypeError):
+                    logger.warning("Invalid border_color specified. Borders will not be displayed.")
```

Comment:
Maybe add "bar" somewhere in this warning.

What's the `TypeError` risk here?

I wonder if `rgb` shouldn't raise `ConfigError` instead and we catch that... Or we could add a new error.. Just thinking out loud, ignore this :)

I wonder if we could `rgb` these colours in `__init__` so that it only needs to be done once. Would that work, or maybe introduce potential problems?

Reply:
I think you're right, `rgb` only raises `ValueError`s.

As for the last point, yes, this check could be done in `__init__`.